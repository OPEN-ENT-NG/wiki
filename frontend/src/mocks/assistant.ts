import {
  PagesContentResponseData,
  PagesStructureResponseData,
} from '~/services/api/assistant/assistant.types';

// ==================================
// Mocked Pages Structure Data
// ==================================
export const mockPagesStructureResponse: PagesStructureResponseData = {
  title: 'Intelligence Artificielle',
  description:
    'Cours généré automatiquement sur le thème: Intelligence Artificielle',
  pages: [
    {
      title: 'Intelligence Artificielle',
      isVisible: true,
      position: 1,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
    },
    {
      title: 'Introduction',
      isVisible: true,
      position: 2,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
    },
    {
      title: "Types d'IA",
      isVisible: true,
      position: 3,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
    },
    {
      title: 'Applications pratiques',
      isVisible: true,
      position: 4,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
    },
  ],
  thumbnail: '',
  created: '2025-10-22T13:28:02.093260',
  modified: '2025-10-22T13:28:02.093260',
  owner: { userId: 'test-user-123', displayName: 'User_test-use' },
};

// ==================================
// Mocked Pages Content Data
// ==================================
export const mockPagesContentResponse: PagesContentResponseData = {
  title: 'Intelligence Artificielle',
  description:
    'Cours généré automatiquement sur le thème: Intelligence Artificielle',
  pages: [
    {
      title: 'Intelligence Artificielle',
      isVisible: true,
      position: 1,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
      content: "Contenu de la page sur l'Intelligence Artificielle...",
    },
    {
      title: 'Introduction',
      isVisible: true,
      position: 2,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
      content: "Contenu de la page d'introduction...",
    },
    {
      title: "Types d'IA",
      isVisible: true,
      position: 3,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
      content: "Contenu de la page sur les types d'IA...",
    },
    {
      title: 'Applications pratiques',
      isVisible: true,
      position: 4,
      author: 'test-user-123',
      authorName: 'User_test-use',
      modified: '2025-10-22T13:28:02.093260',
      created: '2025-10-22T13:28:02.093260',
      lastContributer: 'test-user-123',
      lastContributerName: 'User_test-use',
      content: 'Contenu de la page sur les applications pratiques...',
    },
  ],
  thumbnail: '',
  created: '2025-10-22T13:28:02.093260',
  modified: '2025-10-22T13:28:02.093260',
  owner: { userId: 'test-user-123', displayName: 'User_test-use' },
};
