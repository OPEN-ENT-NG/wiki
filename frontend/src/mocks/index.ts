import { http, HttpResponse } from 'msw';
import { baseURL } from '~/services/api';

export const mockWiki = {
  _id: '6e3d23f6-890f-4453-af19-f9853a14b354',
  title: 'Wiki Clément',
  pages: [
    {
      _id: '6669c22b1b543d54ee3d804e',
      title: 'ma nouvelle page',
      contentPlain: 'test modification',
      author: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      authorName: 'Madame isabelle POLONIO (prof arts plastiques)',
      modified: {
        $date: 1718207454421,
      },
      lastContributer: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      lastContributerName: 'Madame isabelle POLONIO (prof arts plastiques)',
    },
    {
      _id: '6669c3e939d30a7f7e7ada17',
      title: 'aaaa',
      contentPlain: 'rgrgrg aaz',
      author: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      authorName: 'Madame isabelle POLONIO (prof arts plastiques)',
      modified: {
        $date: 1718207659179,
      },
      lastContributer: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      lastContributerName: 'Madame isabelle POLONIO (prof arts plastiques)',
    },
    {
      _id: '6688100c67c6ea0825d056b5',
      title: 'test gras',
      contentPlain: 'ma super page',
      author: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      authorName: 'Madame isabelle POLONIO (prof arts plastiques)',
      modified: {
        $date: 1720193036074,
      },
    },
    {
      _id: '668819aa359b28399237368f',
      title: 'zefzefzefzef',
      contentPlain: 'zefzefzef',
      author: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      authorName: 'Madame isabelle POLONIO (prof arts plastiques)',
      modified: {
        $date: 1720195498738,
      },
    },
  ],
  modified: {
    $date: 1718207659179,
  },
  owner: {
    userId: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
    displayName: 'Madame isabelle POLONIO (prof arts plastiques)',
  },
  index: '668819aa359b28399237368f',
};

export const mockWikis = [
  {
    _id: '6e3d23f6-890f-4453-af19-f9853a14b354',
    title: 'Wiki Clément',
    modified: {
      $date: 1718207659179,
    },
    owner: {
      userId: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      displayName: 'Madame isabelle POLONIO (prof arts plastiques)',
    },
    index: '668819aa359b28399237368f',
  },
  {
    _id: '2d5bcc14-53bd-4e3a-b886-012a993f9b2b',
    title: 'Les recettes provençales — Copie',
    thumbnail: '/workspace/document/f0ee4441-9f40-41ce-9235-8d1ff68a54fe',
    modified: {
      $date: 1716383468143,
    },
    owner: {
      userId: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      displayName: 'Madame isabelle POLONIO (prof arts plastiques)',
    },
    index: '664c580b17cf9a1d353b97d3',
  },
];

export const mockWikisWithPages = [
  {
    _id: '423a3e89-d3a4-4524-97e7-f476c71d9d6e',
    title: 'Définitions Espanol',
    pages: [
      {
        _id: '5540ede1e4b056471c495282',
        title: 'Acaparar',
        author: 'b6db229d-6dae-4949-a405-6ea685c05998',
        authorName: "SANDRINE BOUQUETY (Prof. d'espagnol)",
        modified: {
          $date: 1430318586054,
        },
        contentPlain: '',
      },
      {
        _id: '5540eec6e4b056471c495283',
        title: 'Resultar',
        author: '13f1e9f7-7192-4f75-9a6b-c1c7eefe0280',
        authorName: 'Patrick MOSSION',
        modified: {
          $date: 1501842229907,
        },
        contentPlain: '',
      },
      {
        _id: '5540ef0fe4b056471c495284',
        title: 'Lechería',
        author: 'b6db229d-6dae-4949-a405-6ea685c05998',
        authorName: "SANDRINE BOUQUETY (Prof. d'espagnol)",
        modified: {
          $date: 1430318863835,
        },
        contentPlain: '',
      },
      {
        _id: '580f3f8ee4b07ac1b023b2db',
        title: 'Accueil',
        contentPlain:
          'Quelques définitions des mots vus lors de la dernière séance :&nbsp;&nbsp;- Resultar&nbsp;&nbsp;- Lecheria&nbsp;&nbsp;- Acaparar',
        author: 'a87b39c0-9b23-42db-ade1-7675ae99e898',
        authorName: 'CLAIRE OGOULA-NKWEMI (documentaliste)',
        modified: {
          $date: 1478192151290,
        },
      },
      {
        _id: '657974cc736a590490395b4b',
        title: 'Titre',
        contentPlain: 'texte',
        author: '219a1774-f581-4386-acb3-1a0b06fbd159',
        authorName: 'DEVAULX ALAIN',
        modified: {
          $date: 1702458572178,
        },
      },
    ],
    owner: {
      userId: 'b6db229d-6dae-4949-a405-6ea685c05998',
      displayName: "SANDRINE BOUQUETY (Prof. d'espagnol)",
    },
    modified: {
      $date: 1501842229907,
    },
    thumbnail: '/workspace/document/33d77b15-cba9-41c8-9f83-f4a785752f18',
    shared: [],
    index: '580f3f8ee4b07ac1b023b2db',
  },
];

export const mockPage = {
  _id: '6e3d23f6-890f-4453-af19-f9853a14b354',
  title: 'Wiki Clément',
  pages: [
    {
      _id: '6669c22b1b543d54ee3d804e',
      title: 'ma nouvelle page',
      content:
        '\u003Cdiv class="ng-scope"\u003Etest modification\u003C/div\u003E',
      contentPlain: 'test modification',
      author: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      authorName: 'Madame isabelle POLONIO (prof arts plastiques)',
      modified: {
        $date: 1718207454421,
      },
      lastContributer: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
      lastContributerName: 'Madame isabelle POLONIO (prof arts plastiques)',
    },
  ],
  owner: {
    userId: '91c22b66-ba1b-4fde-a3fe-95219cc18d4a',
    displayName: 'Madame isabelle POLONIO (prof arts plastiques)',
  },
};

export const mockRevision = [
  {
    _id: '235ab109-0421-4251-8693-800740006661',
    wikiId: '6ef1343b-e07d-4bc4-828f-20a300e22a18',
    pageId: '6684fdce6a32ef3e1a62c92d',
    userId: '2875315d-48a2-46fd-b21a-4b6150761730',
    username: 'ADMC Clément',
    title: '22ma new page',
    content: '<div class="ng-scope">update</div>',
    date: {
      $date: 1720431608339,
    },
  },
  {
    _id: 'c7e68db2-88f6-4e1a-bbf5-cf7c8fbce895',
    wikiId: '6ef1343b-e07d-4bc4-828f-20a300e22a18',
    pageId: '6684fdce6a32ef3e1a62c92d',
    userId: '2875315d-48a2-46fd-b21a-4b6150761730',
    username: 'ADMC Clément',
    title: '22ma new page',
    content: '<div class="ng-scope">aaaaaaazezef 222</div>',
    date: {
      $date: 1720184598119,
    },
  },
];

/**
 * MSW Handlers
 */
export const handlers = [
  http.get(`/wiki/list`, () => {
    return HttpResponse.json(mockWikis, { status: 200 });
  }),
  http.get(`/wiki/listallpages`, () => {
    return HttpResponse.json(mockWikisWithPages, { status: 200 });
  }),
  http.get(`/wiki/:wikiId/listpages`, () => {
    return HttpResponse.json(mockWiki, { status: 200 });
  }),
  http.get(`/wiki/:wikiId/page/:pageId`, () => {
    return HttpResponse.json(mockPage);
  }),
  http.get(`${baseURL}/revisions/${mockWiki._id}/:pageId`, () => {
    return HttpResponse.json(mockRevision, { status: 200 });
  }),
  http.post(`/wiki/:wikiId/page`, async ({ request }) => {
    const newPage = await request.json();
    return HttpResponse.json(newPage, { status: 201 });
  }),
  http.put(`/wiki/:wikiId/page/:pageId`, async ({ request }) => {
    return HttpResponse.json({ number: 1 }, { status: 200 });
  }),
  http.delete(`wiki/:wikiId/page/:pageId`, () => {
    return HttpResponse.json({ number: 0 });
  }),
];
