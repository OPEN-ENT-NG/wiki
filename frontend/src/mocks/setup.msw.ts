import { afterAll, afterEach, beforeAll } from 'vitest';
import { server } from './node';

// Enable API mocking before tests.
beforeAll(() =>
  server.listen({
    onUnhandledRequest: 'bypass',
  })
);

// Reset any request handlers that are declared as a part of our tests
// (i.e. for testing one-time error scenarios)
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());
