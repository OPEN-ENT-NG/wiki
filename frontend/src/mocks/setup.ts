/**
 * DO NOT MODIFY
 */

import '@testing-library/jest-dom/vitest';
import { RenderOptions, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ReactElement } from 'react';
import { afterAll, afterEach, beforeAll } from 'vitest';
import '../i18n';
import { CustomProviders } from '../providers';
import { server } from './server';

// Enable API mocking before tests.
beforeAll(() =>
  server.listen({
    onUnhandledRequest: 'bypass',
  }),
);
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

const user = userEvent.setup();

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => {
  return {
    user,
    ...render(ui, { wrapper: CustomProviders, ...options }),
  };
};

export const wrapper = CustomProviders;
export * from '@testing-library/react';
export { customRender as render };
