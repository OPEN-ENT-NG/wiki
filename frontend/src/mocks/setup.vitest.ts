import '@testing-library/jest-dom';
import { RenderOptions, render } from '@testing-library/react';
import { ReactElement } from 'react';

import '../i18n';
import { Providers } from '../providers';
import './setup.msw';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
