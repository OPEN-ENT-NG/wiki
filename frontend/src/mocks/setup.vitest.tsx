import '@testing-library/jest-dom';
import { RenderOptions, render } from '@testing-library/react';
import { ReactElement } from 'react';

import userEvent from '@testing-library/user-event';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import '../i18n';
import { Providers, queryClient } from '../providers';
import './setup.msw';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => {
  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: Providers, ...options }),
  };
};

/**
 * https://reactrouter.com/en/main/routers/create-memory-router
 * We use Memory Router with its "data router" form to make asumptions
 * Useful when we want to test navigation or redirection
 */
export const renderWithRouter = (path = '/', element: JSX.Element) => {
  const routes = [
    {
      path,
      element,
    },
  ];

  const router = createMemoryRouter(routes, {
    initialEntries: [path],
  });

  return {
    /**
     * We use our customRender fn to wrap Router with our Providers
     */
    ...customRender(<RouterProvider router={router} />),
  };
};

export const wrapper = Providers;
export * from '@testing-library/react';
export { queryClient, customRender as render };

beforeEach(() => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: vi.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: vi.fn(), // deprecated
      removeListener: vi.fn(), // deprecated
      addEventListener: vi.fn(),
      removeEventListener: vi.fn(),
      dispatchEvent: vi.fn(),
    })),
  });
});
