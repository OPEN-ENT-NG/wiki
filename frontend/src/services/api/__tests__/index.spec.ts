import { test } from 'vitest';
import {
  mockPage,
  mockRevision,
  mockWiki,
  mockWikis,
  mockWikisWithPages,
} from '~/mocks';
import { server } from '~/mocks/node';
import { wikiService } from '..';

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any request handlers that are declared as a part of our tests
// (i.e. for testing one-time error scenarios)
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

describe('Wiki GET Methods', () => {
  test('makes a GET request to get all wikis without pages', async () => {
    const response = await wikiService.getAllWikis();

    expect(response).toBeDefined();
    expect(response).toHaveLength(2);
    expect(response).toStrictEqual(mockWikis);
    expect(response).toMatchSnapshot();
  });

  test('makes a GET request to get all wikis with pages', async () => {
    const response = await wikiService.getAllWikisWithPages();

    expect(response).toBeDefined();
    expect(response).toStrictEqual(mockWikisWithPages);
  });

  test('makes a GET request to get one wiki with pages', async () => {
    const response = await wikiService.getWiki(mockWiki._id);

    expect(response).toBeDefined();
    expect(response).toStrictEqual(mockWiki);
  });
});

describe('Wiki Page GET Methods', () => {
  test('makes a GET request to get one page of a wiki', async () => {
    const response = await wikiService.getPage({
      wikiId: mockPage._id,
      pageId: mockPage.pages[0]._id,
    });

    expect(response).toBeDefined();
    expect(response).toHaveProperty('_id');
    expect(response).toHaveProperty('pages');
    expect(response).toStrictEqual(mockPage);
  });

  test('makes a GET request to get revisions of a page', async () => {
    const response = await wikiService.getRevisionsPage({
      wikiId: mockPage._id,
      pageId: mockPage.pages[0]._id,
    });

    expect(response).toBeDefined();
    expect(response).toStrictEqual(mockRevision);
  });
});

describe('Wiki Page Mutation Methods', () => {
  test('makes a POST request to create a new page of a wiki', async () => {
    const response = await wikiService.createPage({
      wikiId: mockPage._id,
      data: {
        title: "page d'accueil",
        content: 'test',
      },
    });

    expect(response).toHaveProperty('title');
    expect(response).toHaveProperty('content');
  });

  test('makes a PUT request to update a page of a wiki', async () => {
    const response = await wikiService.updatePage({
      wikiId: mockPage._id,
      pageId: mockPage.pages[0]._id,
      data: {
        title: "page d'accueil",
        content: 'test',
      },
    });

    expect(response).toHaveProperty('number', 1);
  });

  test('makes a DELETE request to delete a page of a wiki', async () => {
    const response = await wikiService.deletePage({
      wikiId: mockPage._id,
      pageId: mockPage.pages[0]._id,
    });

    expect(response).toHaveProperty('number', 0);
  });
});
