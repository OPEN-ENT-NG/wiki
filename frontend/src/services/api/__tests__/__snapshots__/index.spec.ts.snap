// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Wiki GET Methods > makes a GET request to get all wikis without pages 1`] = `
[
  {
    "_id": "6e3d23f6-890f-4453-af19-f9853a14b354",
    "index": "668819aa359b28399237368f",
    "modified": {
      "$date": 1718207659179,
    },
    "owner": {
      "displayName": "Madame isabelle POLONIO (prof arts plastiques)",
      "userId": "91c22b66-ba1b-4fde-a3fe-95219cc18d4a",
    },
    "title": "Wiki Clément",
  },
  {
    "_id": "2d5bcc14-53bd-4e3a-b886-012a993f9b2b",
    "index": "664c580b17cf9a1d353b97d3",
    "modified": {
      "$date": 1716383468143,
    },
    "owner": {
      "displayName": "Madame isabelle POLONIO (prof arts plastiques)",
      "userId": "91c22b66-ba1b-4fde-a3fe-95219cc18d4a",
    },
    "thumbnail": "/workspace/document/f0ee4441-9f40-41ce-9235-8d1ff68a54fe",
    "title": "Les recettes provençales — Copie",
  },
]
`;
